app/javascript/components/schedule-form/schedule-form-fields.js

```diff
@@ -0,0 +1,276 @@
+  id: 'object_type',
+  name: 'target_class',
+  label: __('Type'),
+  onChange: (value) => objectTypeChange(value, setData, data),
+  hideField: data.displayFields.hideAutomationFields,
+  placeholder: __('<Choose>'),
+  includeEmpty: true,
+  options: data.options.objectType,
+  isRequired: true,
+  validate: [{ type: 'customRequired', hideField: data.displayFields.hideAutomationFields }],
+});
+
+export const objectItemField = (data) => ({
+  component: componentTypes.SELECT,
+  id: 'object_item',
+  name: 'target_id',
+  label: __('Object'),
+  hideField: data.displayFields.hideObjectItem,
+  options: data.options.objectItem,
+  placeholder: __('<Choose>'),
+  includeEmpty: true,
+  isRequired: true,
+  validate: [{ type: 'customRequired', hideField: data.displayFields.hideObjectItem }],
+});
+
+const attributeValueField = (count) => ({
+  component: componentTypes.SUB_FORM,
+  id: `attribute-value-field-${count}`,
+  name: `attribute-value-field-${count}`,
+  className: 'attribute-value-field-row',
+  fields: [
+    {
+      id: `attribute-value-count-${count}`,
+      component: componentTypes.PLAIN_TEXT,
+      name: `attribute-value-count-${count}`,
+      label: count.toString(),
+      className: 'attribute-value-row-count',
+    },
+    {
+      component: componentTypes.TEXT_FIELD,
+      id: `attribute_${count}`,
+      name: `attribute_${count}`,
+      label: __(' '),
+      className: 'attribute-value-row-attribute',
+    },
+    {
+      component: componentTypes.TEXT_FIELD,
+      id: `value_${count}`,
+      name: `value_${count}`,
+      label: __(' '),
+      className: 'attribute-value-row-value',
+    },
+  ],
+}
+);
+
+export const attributeValueFields = (data, limit) => ({
+  component: componentTypes.SUB_FORM,
+  id: `attribute-value-fields`,
+  name: `attribute-value-fields`,
+  className: 'attribute-value-fields-subform',
+  title: __('Attribute/Value Pairs'),
+  condition: {
+    when: 'action_typ',
+    is: 'automation_request',
+  },
+  fields: [[...Array(limit)].map((_item, i) => attributeValueField(i + 1)),
+    {
+      component: componentTypes.TEXT_FIELD,
+      id: `starting_object`,
+      name: `starting_object`,
+      label: __(' '),
+      value: 'SYSTEM/PROCESS',
+      hideField: true,
+    },
+  ],
+});
+
+export const runField = (setData, data) => ({
+  component: componentTypes.SELECT,
+  id: 'run',
+  name: 'timer_typ',
+  label: __('Run'),
+  initialValue: 'Once',
+  isRequired: true,
+  onChange: (value) => runOptionChange(value, setData, data),
+  options: runOptions(),
+});
+
+export const timerValueField = (data) => ({
+  component: componentTypes.SELECT,
+  id: 'timer_value',
+  name: 'timer_value',
+  label: __('Every'),
+  hideField: data.displayFields.hideEveryTime,
+  options: data.options.everyTime,
+  isRequired: true,
+  initializeOnMount: true,
+  initialValue: data.initialValues.timer_value || 1,
+});
+
+export const timezoneField = (data) => ({
+  component: componentTypes.SELECT,
```

Comment:
There are a lot of timezones here which makes navigating this select kind of difficult. In other areas of the ui we tend to use a searchable select to list timezones which I think would also make a lot of sense here (See: https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/javascript/components/visual-settings-form/visual-settings-form.schema.js#L104-L112).

![image](https://user-images.githubusercontent.com/64800041/217944259-79fb09be-7d0d-4161-8f97-17a10233f78c.png)

Reply:
added a search feature to the select box.